### ISSUE TITLE
SetMusic() forces AutoPlay to true

### ISSUE THREAD

**Issue by ericdrobinson** [MoePlayer/APlayer#170]
## Description
Calling `setMusic` on a paused player will automatically play the music, even with `autoplay=false`.

## The Steps to Reproduce
1) Load music into APlayer instance.
2) Play music.
3) Stop music.
4) Change music loaded into APlayer (e.g. `ap.option.music[0] = {...}`).
5) Call `setMusic()` (e.g. `ap.setMusic(0)`) to update the appearance of the player.

**Expected Behavior:** The player should reload the information (including metadata, if `autoplay="metadata"` was set) but should not begin playback.
**Actual Behavior:** The player automatically begins playback.

## Demo of the Problem
Please see [this JSFiddle](https://jsfiddle.net/4x417cg5/) for an example. Do the following to see it:
1) Press the Play button.
2) Press the Pause button.
3) Press the custom Swap button.

I don't know why but the audio in the JSFiddle also won't change. All other information properly changes but the loaded music does not update. Perhaps this is another bug?

## Version Information
**APlayer:** 1.6.0
**Browsers:** All

## Extra Information
This is likely caused by setting `this.option.autoplay = true;` [at the end of the `setMusic` function](https://github.com/MoePlayer/APlayer/blob/77d00898bec67ff9f9b07ef5bc3fb09061483cac/src/APlayer.js#L729). At the very least, that line should be guarded by a check for playlist, rather than a single loaded audio file. Another option would be to check the `mode` option for `circulation` or `order`.

**Comment by MPierre9** [comment]
I think it's a problem with autoplay being set to true like @ericdrobinson said. I removed it and instead checked if the current song is paused or not to determine whether the player should remain paused or play the song being set.

## Code
```
 if(!this.element.getElementsByClassName('aplayer-button aplayer-play')[0]) { // if current song is playing, play song being set 
            this.play();
        }
        else { // if current song is paused, pause the song being set 
            this.pause();
        }
```

## Result

![autoplay-possible-fix](https://user-images.githubusercontent.com/27080760/33898420-5727209e-df36-11e7-95a9-33d1562c2792.gif)

**Comment by ericdrobinson** [comment]
Hmm... @MPierre9 where did you add that code? I'm looking at it and thinking that the check for autoplay should be added to the constructor after `setMusic()` is called (right before "`instances.push(this)`". The `autoplay` value really only makes sense for the first call. After that, you can check in the `ended` callback where looping/advancing/etc. are handled. In that case, simply call "`this.play()`" after the call to `setMusic` in the sections where you expect music to keep playing. (I _think_ it's safe to assume that the audio would have been playing if you receive the "`ended`" callback...)

**Comment by MPierre9** [comment]
I added it [here](https://github.com/MoePlayer/APlayer/blob/master/src/APlayer.js#L729) at the end of `setMusic()`. In the APlayer demo for "With playlist" autoplay is set to false so it will pause if the `setMusic()` is called. Refined code would be: 
```
// autoplay 
        if(!this.element.getElementsByClassName('aplayer-button aplayer-play')[0] && this.option.autoplay && !this.isMobile) { // if current song is playing, play song being set 
            this.play();
        }
        else { // if current song is paused, pause the song being set 
            this.pause();
        }
```

Gives the same result. Your idea makes sense however it seems like extra code that should be handled inside `setMusic`. Do you have some example implementation?

**Comment by ericdrobinson** [comment]
I'm working on a fix for this right now, actually, and will issue a PR soon.

The key insight is that `autoplay` is being used for **multiple** purposes:

1) Whether to start the audio playback when the page loads.
2) Whether to continue playing in a list context.

Whoops. Changing one affects the other! Rather, `autoplay` should _only_ be checked during initialization (the constructor). Then, the second situation I outlined above can be handled in the code that calls `setMusic`, [here](https://github.com/MoePlayer/APlayer/blob/master/src/APlayer.js#L566). Wherever it makes sense, you simply add a call to `this.play()` after `setMusic` and it's effectively the same thing (albeit safer).

In the case that someone clicks on a music entry in a playlist (whether the player is playing or stopped), the code _already_ handles this by assuming the intention is to play the music ([see this line](https://github.com/MoePlayer/APlayer/blob/master/src/APlayer.js#L260)).

That said, it would be useful to add an API for determining whether the APlayer instance was playing music or not. In that case, however, I'd suggest adding a member function like this:

```js
isPlaying() {
    return !this.audio.paused();
}
```
That would be faster than checking the state of the DOM...

**Comment by ericdrobinson** [comment]
Recommend closing this issue as it was fixed by #174.
